### PR TITLE
fix processor

### PR DESCRIPTION
**Pull request by hindnasser**

### CODE REVIEW

File: src/main/java/com/apollo/resource/controller/ResourceController.java [hunk 1]
```diff
@@ -17,21 +17,24 @@
...
+        return this.resourceService.getResourceByID(resourceId).map(Optional::get);
+    }
 
     @PostMapping(value = "/")
     public Mono<Resource> createResource(@RequestBody Resource resource){
         return this.resourceService.postResource(Mono.just(resource));//try
     }
 
-    @PutMapping(value= "/resourceId")
+    @PutMapping(value= "/{resourceId}")
```
Comment (MrMoon) [inline src/main/java/com/apollo/resource/controller/ResourceController.java:30]: no need for the {resourceId}

File: src/main/java/com/apollo/resource/controller/ResourceController.java [hunk 2]
```diff
@@ -17,21 +17,24 @@
...
     public Mono<Boolean> updateResource(@RequestBody Resource resource){
         return this.resourceService.updateResource(Mono.just(resource));
     }
 
     @PutMapping(value = "/share/{flag}")
-    public Mono<Boolean> shareResource (@PathVariable ("flag") Boolean flag/*,@RequestBody String userId, String resourceId*/){
-        return null;
-        //return this.resourceService.shareResourceFlag(userId,resourceId,flag);
+    public Mono<Boolean> shareResource (@PathVariable ("flag") Boolean flag ,@RequestBody String userId, String resourceId){
```
Comment (MrMoon) [inline src/main/java/com/apollo/resource/controller/ResourceController.java:36]: resourceId should also be @RequestBody, so create an object the have userId and resourceId and just @RequstBody once

File: src/main/java/com/apollo/resource/model/Document.java [hunk 3]
```diff
@@ -11,9 +11,7 @@
 @Data
 public class Document extends Resource {
 
-    private final String documentId = UUID.randomUUID().toString();
-    private final Date documentDateOfCreation = Calendar.getInstance().getTime();
-    private String documentName = this.getResourcesOwnerId() + "-" + documentDateOfCreation;// how to get owner's name
-    private String documentUrl, OwnerId = this.getResourcesOwnerId();//check this
+    private String documentName = this.getResourcesOwnerId() + "-" + getResourceCreationDate();
+    private String documentUrl, OwnerId = this.getResourcesOwnerId();
```
Comment (MrMoon) [inline src/main/java/com/apollo/resource/model/Document.java:11]: OwnerId should be ownerId, camelCase convention

**Review by MrMoon — APPROVED** [review]

**Review by MrMoon — APPROVED** [review]

File: src/main/java/com/apollo/resource/model/ResourceType.java [hunk 4]
```diff
@@ -0,0 +1,5 @@
+package com.apollo.resource.model;
+
+public enum ResourceType {
+    Document, Video
```
Comment (MrMoon) [inline src/main/java/com/apollo/resource/model/ResourceType.java:4]: The naming convention for the enums are ALLCAPS
So,
DOCUMENT, VIDEO

**Review by MrMoon — APPROVED** [review]

File: src/main/java/com/apollo/resource/model/ResourceType.java (hunk 4 above)
Comment (hindnasser) [reply src/main/java/com/apollo/resource/model/ResourceType.java]: Done

**Review by MrMoon — APPROVED** [review]

**Review by MrMoon — APPROVED** [review]

File: src/main/resources/application.yml [hunk 5]
```diff
@@ -31,8 +31,8 @@ spring:
           binder:
             application-id: apollo-resource
       bindings:
-        courseProcessor-in-0.destination: resource
-        courseProcessor-out-0.destination: resource-state-store
+        resourceProcessor-in-0.destination: resource
+        resourceProcessor-out-0.destination: resource-state-store
 
 server:
   port: 8081
```
Comment (MrMoon) [inline src/main/resources/application.yml:38]: port 8081 is already taken by the course microservice, make it 8084

**Review by MrMoon — CHANGES_REQUESTED** [review]

**Review by MrMoon — APPROVED** [review]

**Pull request merged**